### PR TITLE
fix: [OIP-753]: Filter only the points within the given duration

### PR DESCRIPTION
**Pull request by praneshg239**
#### Screenshots

Before 

image

After

image

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.utils.ts [hunk 1]
```diff
@@ -168,7 +174,11 @@ function generatePointsForNodes({
   const testDataInitialXValue = sortedTestData?.[0]?.timestampInMillis || 0
 
   sortedTestData?.forEach(({ timestampInMillis, value }) => {
-    hostData.push({ x: (timestampInMillis || 0) - startTime, y: value === -1 ? null : value })
+    const xValue = (timestampInMillis || 0) - startTime
+
+    if (xValue < durationInMinutes * 60000) {
```
Comment (harshilbits) [inline src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.utils.ts:179]: This if condition can be moved to separate function with the name explaining what exactly it does. We can also add A UT for the same.

Comment (praneshg239) [reply src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.utils.ts]: Done

File: src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.tsx [hunk 2]
```diff
@@ -94,13 +94,16 @@ export function DeploymentMetrics(props: DeploymentMetricsProps): JSX.Element {
...
+  const { controlDataStartTimestamp, testDataStartTimestamp, spec: overviewSpec } = overviewData || {}
+
+  const isSimpleVerification = overviewSpec?.analysisType === VerificationJobType.SIMPLE
 
   const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()
 
   const startTimestampData: StartTimestampDataType = {
-    controlDataStartTimestamp: overviewData?.controlDataStartTimestamp || 0,
-    testDataStartTimestamp: overviewData?.testDataStartTimestamp || 0
+    controlDataStartTimestamp: controlDataStartTimestamp || 0,
```
[the remark this answers is not in the document]
Comment (praneshg239) [reply src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.tsx:104]: if the value is undefined, then we need to default assign 0. So we have to use this pattern

File: src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.tsx [hunk 3]
```diff
@@ -94,13 +94,16 @@ export function DeploymentMetrics(props: DeploymentMetricsProps): JSX.Element {
   const { getString } = useStrings()
   const pageParams = useQueryParams<ExecutionQueryParams>()
 
-  const isSimpleVerification = overviewData?.spec?.analysisType === VerificationJobType.SIMPLE
+  const { controlDataStartTimestamp, testDataStartTimestamp, spec: overviewSpec } = overviewData || {}
```
Comment (deepesh-ui) [inline src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.tsx:97]: const {
  controlDataStartTimestamp = 0,
  testDataStartTimestamp = 0,
  spec: overviewSpec = {}
} = overviewData || {};

Comment (praneshg239) [reply src/modules/85-cv/components/ExecutionVerification/components/DeploymentMetrics/DeploymentMetrics.tsx]: Done

**Review by deepesh-ui — APPROVED** [review]

**Review by karansaraswat19 — APPROVED** [review]

**Pull request merged**